src/EventHandler/AbstractButtonQuery.php

```diff
@@ -6,8 +6,13 @@
 
 abstract class AbstractButtonQuery extends AbstractQuery
 {
+    /** @var string Data associated with the callback button. Be aware that a bad client can send arbitrary data in this field. */
+    public readonly string $data;
+
+    /** @internal */
     public function __construct(MTProto $API, array $rawCallback)
     {
         parent::__construct($API, $rawCallback);
+        $this->data = (string) $rawCallback['data'];
```

Comment:
No need to cast this

Reply:
i say my reason and my exaple in your telegram pv so i close this conversation as resolved